crates/polars-ops/src/chunked_array/strings/namespace.rs

```diff
@@ -115,23 +115,26 @@ pub trait Utf8NameSpaceImpl: AsUtf8 {
             },
             _ => {
                 if literal {
-                    Ok(binary_elementwise_values(ca, pat, |src, pat| {
+                    Ok(broadcast_binary_elementwise_values(ca, pat, |src, pat| {
                         src.contains(pat)
                     }))
                 } else if strict {
                     // A sqrt(n) regex cache is not too small, not too large.
                     let mut reg_cache = FastFixedCache::new((ca.len() as f64).sqrt() as usize);
-                    try_binary_elementwise(ca, pat, |opt_src, opt_pat| match (opt_src, opt_pat) {
-                        (Some(src), Some(pat)) => {
-                            let reg = reg_cache.try_get_or_insert_with(pat, |p| Regex::new(p))?;
-                            Ok(Some(reg.is_match(src)))
-                        },
-                        _ => Ok(None),
+                    broadcast_try_binary_elementwise(ca, pat, |opt_src, opt_pat| {
```

Comment:
This is the one case where I'd rather have the broadcasting explicit in the implementation instead of delegating with `broadcast_*`. Broadcasting the pattern is very common, and it means we don't have to do a cache lookup each iteration.